### PR TITLE
Refactor BossBar management

### PR DESCRIPTION
**Pull request by Pr0methean**
Was previously exposing BossBarManager's own list field to external modification.

### CODE REVIEW

**Comment by Pr0methean** [comment]
Totally rewritten following discussion with kamcio96 on the Discord.

File: src/main/java/net/glowstone/boss/BossBarManager.java [hunk 1]
```diff
@@ -1,83 +1,49 @@
...
-        }
-        this.server = server;
-        instance = this;
-    }
-
-    public static BossBarManager getInstance() {
-        return instance;
-    }
-
+public enum BossBarManager {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/boss/BossBarManager.java]: Please make this a `static class`.

File: src/main/java/net/glowstone/boss/BossBarManager.java [hunk 2]
```diff
@@ -1,83 +1,49 @@
...
+    ;
```
Comment (mastercoms) [inline src/main/java/net/glowstone/boss/BossBarManager.java]: One lonely semicolon, who needs to be destroyed.

**Review by mastercoms — DISMISSED** [review]

File: src/main/java/net/glowstone/boss/BossBarManager.java (hunk 1 above)
Comment (Pr0methean) [reply src/main/java/net/glowstone/boss/BossBarManager.java]: Done; some authors recommend that utility classes that aren't going to be instantiated be zero-instance enums.

**Pull request merged**